sparc.patch by Jurij Smakov <jurij@wooyd.org>

(1) Defines ARCH=ultrasparc for any sparc* architecture. This allows
    to build HiPE enabled package for sparc in Debian.

(2) Fixes FTBFS(see http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=328031).

--- erlang-15.b.1-dfsg.orig/erts/configure.in
+++ erlang-15.b.1-dfsg/erts/configure.in
@@ -609,7 +609,7 @@
 fi
 
 case $chk_arch_ in
-    sun4u)	ARCH=ultrasparc;;
+    sun4u|sparc*)	ARCH=ultrasparc;;
     sparc64)	ARCH=sparc64;;
     sun4v)	ARCH=ultrasparc;;
     i86pc)	ARCH=x86;;
--- erlang-15.b.1-dfsg.orig/erts/emulator/beam/beam_emu.c
+++ erlang-15.b.1-dfsg/erts/emulator/beam/beam_emu.c
@@ -1031,16 +1031,6 @@
  * in registers.
  */
 
-#if defined(__GNUC__) && defined(sparc) && !defined(DEBUG)
-#  define REG_x0 asm("%l0")
-#  define REG_xregs asm("%l1")
-#  define REG_htop asm("%l2")
-#  define REG_stop asm("%l3")
-#  define REG_I asm("%l4")
-#  define REG_fcalls asm("%l5")
-#  define REG_tmp_arg1 asm("%l6")
-#  define REG_tmp_arg2 asm("%l7")
-#else
 #  define REG_x0
 #  define REG_xregs
 #  define REG_htop
@@ -1049,7 +1039,6 @@
 #  define REG_fcalls
 #  define REG_tmp_arg1
 #  define REG_tmp_arg2
-#endif
 
 #ifdef USE_VM_PROBES
 #  define USE_VM_CALL_PROBES
